### PR TITLE
[5.8-wip] Add support for nullable unique indexes in SQL Server

### PR DESCRIPTION
**Pull request by jlsjonas**
see https://stackoverflow.com/a/767702/620141
Note: Supports SQL Server 2008 & up (11 years, should be safe)

### CODE REVIEW

**Comment by staudenmeir** [comment]
This is a breaking change for people who except the index not to accept multiple `null` values.
You should target Laravel 5.8

You also have to consider indexes with multiple columns.

File: src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php [hunk 1]
```diff
@@ -105,10 +105,11 @@ public function compilePrimary(Blueprint $blueprint, Fluent $command)
      */
     public function compileUnique(Blueprint $blueprint, Fluent $command)
     {
-        return sprintf('create unique index %s on %s (%s)',
+        return sprintf('create unique index %s on %s (%s) where %s is not null',
             $this->wrap($command->index),
             $this->wrapTable($blueprint),
-            $this->columnize($command->columns)
+            $this->columnize($command->columns),
+            $this->wrap($command->index)
```
Comment (staudenmeir) [inline src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php]: You have to insert the column name(s), not the index name.

Comment (jlsjonas) [reply src/Illuminate/Database/Schema/Grammars/SqlServerGrammar.php]: Yeh, I realized 😅 
I'll retarget, but it'll be something for next week as I have some more pressing issues & also some further local testing (telescope's migrations failed; I'll check asap)

**Comment by driesvints** [comment]
@jlsjonas feel free to resend to master once you're ready.

**Pull request closed without merging**